packages/eas-cli/src/credentials/ios/appstore/ensureAppExists.ts

```diff
@@ -1,24 +1,78 @@
+import { BundleId, CapabilityType, CapabilityTypeOption, Session, Teams } from '@expo/apple-utils';
 import ora from 'ora';
 
-import { AuthCtx } from './authenticate';
+import { AuthCtx, authenticateAsync } from './authenticate';
+import { USE_APPLE_UTILS } from './experimental';
 import { runActionAsync, travelingFastlane } from './fastlane';
 
+export interface EnsureAppExistsOptions {
+  enablePushNotifications?: boolean;
+}
+
 export interface AppLookupParams {
   accountName: string;
   projectName: string;
   bundleIdentifier: string;
 }
 
-export interface EnsureAppExistsOptions {
-  enablePushNotifications?: boolean;
+export async function ensureAuthenticatedAsync(
+  appleCtx: Omit<AuthCtx, 'fastlaneSession'>
+): Promise<Omit<AuthCtx, 'fastlaneSession'>> {
+  if (!Session.getSessionInfo()) {
```

Comment:
I have the same problem here as with `setSelectedTeamId`, I don't think that those values should be hidden inside the library

Reply:
it's pretty standard for an authentication package to manage its own state. These values aren't hidden as they can be retrieved using `Session.getSessionInfo()`. Parts of the auth state contains values related to the network request like cookies, they shouldn't be set externally, and in the singular hacky case when they can be modified it's only to accommodate the ill-formatted yaml cookies that Fastlane uses.